addons/mrp/models/mrp_production.py

```diff
@@ -510,8 +510,14 @@ def write(self, vals):
                     raise UserError(_('You cannot move a manufacturing order once it is cancelled or done.'))
                 if production.workorder_ids and not self.env.context.get('force_date', False):
                     raise UserError(_('You cannot move a planned manufacturing order.'))
-            if 'move_raw_ids' in vals and production.state != 'draft':
-                production.move_raw_ids.filtered(lambda m: m.state == 'draft')._action_confirm()
+            if 'move_raw_ids' in vals:
+                if production.state == 'draft':
```

Comment:
I would do this here https://github.com/odoo-dev/odoo/blob/b546571c6987637fb40f85be950971984abf011c/addons/mrp/models/stock_move.py#L123-L132
The default get already take care of additionnal moves